### ISSUE TITLE
Some apps do not follow Mica's global rules

### ISSUE THREAD

**Issue by VitalSkib** [MicaForEveryone/MicaForEveryone#213]
Hello. 
I have a small problem when some apps do not follow Mica's global rules. 
In my case it's Cinema 4D 2023.1.3. 
It all works fine with the previous Mica version (v1.3.0.0-preview from November 15?), but after installing v1.3.0.0-**preview3**, the Cinema4D title bar remains default white. 
I'm not sure about all apps, but most of them work well, except Cinema 4D (it's just what I noticed right away).
I tried different rules variations, tried to reset the config file, tried different installation options, but none of this helps.
I have to return to the previous version, which works well, as I've already said..
Any ideas why this can happen?
tia

OS: Windows 11 Pro Insider Preview 22H2 25267.1000

**Comment by xmine64** [comment]
* Maybe the app you use has custom titlebar, MFE only works on system titlebar
* Maybe you are running that app as Administrator, in that case you need to run MFE as Administrator.

**Comment by VitalSkib** [comment]
> 

Yes, but it works fine with the previous MFE version (v1.3.0.0-preview, not preview3) and with exactly the same settings. 
That's why I thought that you changed something in the new build.

**Comment by xmine64** [comment]
> That's why I thought that you changed something in the new build.

There's no new change in rule matching. Could you confirm it's not your app's problem?

**Comment by dongle-the-gadget** [comment]
Have you tried v1.3.0.0-preview2? Just trying to narrow down the problem.

**Comment by VitalSkib** [comment]
> Have you tried v1.3.0.0-preview2? Just trying to narrow down the problem.

Yes, and it behaved just like preview3, I just did not mention it..

**Comment by dongle-the-gadget** [comment]
Oh alright. So it happened somewhere between Preview 1 and Preview 2.

**Comment by dongle-the-gadget** [comment]
(wait preview 2 isn't a real tag lol)

**Comment by dongle-the-gadget** [comment]
Well actually due to the problem being so isolated, I would recommend getting a Debug build (but that requires you to also attach a debugger to the program) and doing a rule refresh.

**Comment by VitalSkib** [comment]
> Well actually due to the problem being so isolated, I would recommend getting a Debug build (but that requires you to also attach a debugger to the program - i.e. Visual Studio - rather heavy though) and doing a rule refresh.
> 
> Send us whatever shows up in the output that correlates to the program applying rules.

Well, sorry, I'm not so technically prepared, I am a designer but not a programmer. 
And at the moment I just do not have enough time to study how to do it.
And again, I'm ok whith the previous build for now, if there is no obvious solution.
So, thanks again and sorry for the time spent.

**Comment by xmine64** [comment]
I guess This could be related to #216.

**Comment by VitalSkib** [comment]
I haven't checked it since then, but today I reinstalled it again using the latest installer (dated January 03, 2023) and now it works well for me.. 
The Cinema 4D now also follows the rules and its title bar is now also dark. 
I don’t know what you have changed, but thanks.. 😎👍

**Comment by VitalSkib** [comment]
PS Oops, looks like I was hurrying to rejoice. 
For some reason, it works only partially, some floating windows шт Cinema 4D still retain a white titlbar. 
Weird..
Well, have to return to the working version.

![2023_01_06_Cinema_4D_2023 1 3](https://user-images.githubusercontent.com/43848892/211019944-5ae5639a-d760-47da-b671-5e5057a165bf.jpg)

**Comment by dongle-the-gadget** [comment]
Looks like the window got excluded in one of the new checks that we added. Can you use Spy++ and determine the styles of the window?

**Comment by VitalSkib** [comment]
> Spy++

Oh, sorry, I can’t do it, because I just don't know how to do it. 
As I wrote above, I am a designer and very far from software engineering..

**Comment by matthewadamsdk** [comment]
Hi - I've had this problem as well - just for another app, JPEGView. No matter what settings I choose for Mica, the title bar would always be white.

I think I' ve found the answer!

For Windows 11, select Start  > Settings > Personalization > Colors

Here there's an option to "Show accent colors on tile bars and windows borders". If this ISN'T selected, then JPEGView would always have a light title bar, even though the system color setting was set to "Dark". After selecting this option, JPEGView had a dark bar.

Maybe this will also work for Cinema 4D...?

Which brings me to think: you mentioned that Mica worked fine with the v1.3.0.0-preview version from November 15. That could mean that either Win11 has changed the way the option "Show accent colors..." affects the system UI, or it could mean that that particular version of Mica overruled this option. Maybe @xmine64  could investigate whether the latest version overrules the "Show accent colors..." option, or could be made to do so...?

**Comment by matthewadamsdk** [comment]
PS: if choosing the "Show accent colors..." option doesn't show the *correct* dark color in the title bar, then you could use (for example) the app [Just Color Picker](https://annystudio.com/software/colorpicker/) to select the correct title bar color from another window and enter this RGB code under manual-select in Settings > Personalization > Colors.

**Comment by VitalSkib** [comment]
@matthewadamsdk
Matthew, thank you for trying to help. 
But for me it now works well again, after I installed it using the installer from January 16, I believe. 
I just decided to test it a bit and did not write right away. 
And yes, in my case, it had nothing to do with Windows accent colors and related settings.

**Comment by matthewadamsdk** [comment]
Great to hear 👍

**Comment by VitalSkib** [comment]
FYI
Ok, I give up and return to the old version (dated November 15, 2022). 
Although everything is fine now with my problem in Cinema 4D, I noticed another issue..
I have a "Gmail Notifier" app installed and for some reason its flyout notification window ceased to follow the rules in newer MFE versions.
And instead of rounded corners, I have square one. 
I tried different settings options, resetting and reinstalling already, none of this helps.
However, thanks for your hard work and possible help. 😉👍

![2022_11_15_MFE](https://user-images.githubusercontent.com/43848892/214030034-48925b13-d8f5-4008-90be-bbcae57e35b1.jpg)

![2023_01_18_MFE](https://user-images.githubusercontent.com/43848892/214030082-b9a3eeae-95c0-4dac-ba57-f1c18b8bf4a3.jpg)
